sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureFileSearchAsyncTest.java

```diff
@@ -122,4 +127,37 @@ public void basicFileSearch(HttpClient httpClient, AssistantsServiceVersion serv
                 .verifyComplete();
         });
     }
+
+    @Disabled("file_search tools are not supported in Azure")
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.openai.assistants.TestUtils#getTestParameters")
+    public void fileSearchWithMaxNumberResult(HttpClient httpClient, AssistantsServiceVersion serviceVersion) {
+        client = getAssistantsAsyncClient(httpClient, serviceVersion);
+
+        fileSearchWithMaxNumberResultRunner((fileDetails, assistantCreationOptions) -> {
+            // Upload file
+            StepVerifier.create(client.uploadFile(fileDetails, FilePurpose.ASSISTANTS)
+                .flatMap(openAIFile -> {
+                    // Create assistant
+                    AsyncUtils cleanUp = new AsyncUtils();
+                    CreateToolResourcesOptions createToolResourcesOptions = new CreateToolResourcesOptions();
+                    createToolResourcesOptions.setFileSearch(
+                        new CreateFileSearchToolResourceOptions(
+                            new CreateFileSearchToolResourceVectorStoreOptionsList(
+                                Arrays.asList(new CreateFileSearchToolResourceVectorStoreOptions(
+                                    Arrays.asList(openAIFile.getId()),
+                                    null
```

Comment:
I think that it may be good to add convenience constructors that only accept either `file_ids` or `vector_store_ids`, just so that users don't need to pass `null` if they just want to use one or the other.